### PR TITLE
Campaigns Alerts 

### PR DESCRIPTION
**Pull request by mauditecandela**
- Creates a new alerts handler to include the alerts by campaign (not finished)
- Creates a new method in the campaigns controller that returns an array of alerts ids
- Adds the new alerts handler to the react router
- Includes locales

### CODE REVIEW

File: app/controllers/campaigns_controller.rb [hunk 1]
```diff
@@ -64,6 +64,15 @@ def users
     ).includes(:user, :course).order(revision_count: :desc)
   end
 
+  def alerts
```
Comment (mauditecandela) [inline app/controllers/campaigns_controller.rb]: @ragesoss I would like to have your opinion here. I created a new method in the campaigns controller that returns an array with alerts method. This way we'll be able to call sth like "this.props.campaign.alerts" in the alerts_handler (react). What do you think of this approach?

Thanks in advance!

Comment (ragesoss) [reply app/controllers/campaigns_controller.rb]: This makes sense, but you want to be able to do something like @campaign.alerts, which should come strictly from scopes.

File: app/controllers/campaigns_controller.rb [hunk 2]
```diff
@@ -67,10 +67,10 @@ def users
   def alerts
     alerts = []
     @campaign.course_ids.each do |course_id|
-      alert = Alert.find_by(course_id: course_id)
-      alerts << alert.id unless alert.nil?
+      alert = Alert.where(course_id: course_id)
+      alert.each { |alert| alerts << alert.id unless alert.nil? }
     end
-    alerts
+    render json: { campaign_alerts: alerts }
```
Comment (mauditecandela) [inline app/controllers/campaigns_controller.rb]: @ragesoss do you mean by scope adding it to the campaign model? That makes sense, yes. I was also thinking about adding a reducer and action and therefore creating here a json so we are able to call something like /campaigns/:campaign_slug/alerts.json and receive a json with the information. 
I am not sure what would be better specially in terms of performance or good practices. Any suggestion?

**Comment by mauditecandela** [comment]
captura de pantalla 2018-03-05 a las 18 13 39
@ragesoss this is the current state of the implementation. I would like to have your opinion on both technical and the solution to continue :) thanks!

**Comment by ragesoss** [comment]
The code looks clean and makes sense so far!

**Comment by mauditecandela** [comment]
@ragesoss I modified a bit the outlook. So far it looks it passes al tests. 
I don't know but the "type" attribute from alerts is not being rendered to the json. Do you know why is that happening?

**Comment by ragesoss** [comment]
Cool!

alerts

To be useful, this will need to include article (when there is one) as well. The course, username, and article should all be links. I think 'resolved' should not be shown here. What it means is not very intuitive — it's a feature designed mainly for Wiki Ed staff.

**Pull request merged**